authorization-server/mongodb/clients.js

```diff
@@ -1,75 +1,68 @@
- *     without the user having to make a decision to allow or disallow the scope access.
+ *
+ * trustedClient: default if missing is false. If this is set to true then the client is regarded
+ * as a trusted client and not a 3rd party application. That means that the user will not be
+ * presented with a decision dialog with the trusted application and that the trusted application
+ * gets full scope access without the user having to make a decision to allow or disallow the scope
+ * access.
  */
-var clients = [
-  {
-    id: '1',
-    name: 'Samplr',
-    clientId: 'abc123',
-    clientSecret: 'ssh-secret'
-  },
-  {
-    id: '2',
-    name: 'Samplr2',
-    clientId: 'xyz123',
-    clientSecret: 'ssh-password'
-  },
-  {
-    id: '3',
-    name: 'Samplr3',
-    clientId: 'trustedClient',
-    clientSecret: 'ssh-otherpassword',
-    trustedClient: true
-  }
-];
+const clients = [{
+  id            : '1',
+  name          : 'Samplr',
+  clientId      : 'abc123',
+  clientSecret  : 'ssh-secret',
+}, {
+  id            : '2',
+  name          : 'Samplr2',
+  clientId      : 'xyz123',
+  clientSecret  : 'ssh-password',
+}, {
+  id            : '3',
+  name          : 'Samplr3',
+  clientId      : 'trustedClient',
+  clientSecret  : 'ssh-otherpassword',
+  trustedClient : true,
+}];
 
 /**
  * Returns a client if it finds one, otherwise returns
  * null if a client is not found.
- * @param id The unique id of the client to find
- * @param done The function to call next
- * @returns The client if found, otherwise returns null
+ * @param   {String}   id   - The unique id of the client to find
+ * @param   {Function} done - The client if found, otherwise returns null
+ * @returns {undefined}
  */
-exports.find = function (id, done) {
-  for (var i = 0, len = clients.length; i < len; i++) {
-    var client = clients[i];
-    if (client.id === id) {
-      return done(null, client);
-    }
+exports.find = (id, done) => {
+  const foundClient = clients.find(client => client.id === id);
+  if (foundClient != null) {
+    return done(null, foundClient);
   }
   return done(null, null);
 };
 
 /**
  * Returns a client if it finds one, otherwise returns
  * null if a client is not found.
- * @param clientId The unique client id of the client to find
- * @param done The function to call next
- * @returns The client if found, otherwise returns null
+ * @param   {String}   clientId - The unique client id of the client to find
+ * @param   {Function} done     - The client if found, otherwise returns null
+ * @returns {undefined}
  */
-exports.findByClientId = function (clientId, done) {
-  for (var i = 0, len = clients.length; i < len; i++) {
-    var client = clients[i];
-    if (client.clientId === clientId) {
-      return done(null, client);
-    }
+exports.findByClientId = (clientId, done) => {
+  const foundClient = clients.find(client => client.clientId === clientId);
+  if (foundClient != null) {
```

Comment:
I know you're trying to retain backwards compatibility between undefined vs returning null but since the rest of the code is now using != null which works with both undefined and null, I would go ahead and directly return the results of the find and let it be undefined if it is not found.